76-lynx-boreal/technical_specification.md

```diff
@@ -0,0 +1,553 @@
+# Upload Service Intermediary Revision (Lynx Boreal)
+**Epic Type:** Implementation Epic
+
+Epic planning and implementation follow the
+[Epic Planning and Marathon SOP](https://docs.ghga-dev.de/main/sops/sop001_epic_planning.html).
+
+## Scope
+### Outline:
+The goal of this epic is to overhaul the Upload Controller Service (UCS) as part of the
+new [File Upload concept](https://ghga.pages.hzdr.de/internal.ghga.de/feature_archconcept-file-upload/developer/architecture_concepts/ac007_file_upload/).
+While this concept includes references to future services like the Study Repository Service,
+some of the machinery there is still in the planning phase. This epic will therefore
+rely upon some shortcuts, such as extra Data Steward involvement. When the full
+architectural concept is realized, the appropriate adaptations will follow in a future
+epic.
+
+### Included/Required:
+- Implement new Upload Orchestrator Service as described below
+- Revamp existing UCS logic
+- Adapt WPS for "upload" work packages and tokens
+- Adapt CRS to also manage claims for upload boxes
+- Add new schemas to `ghga-event-schemas`
+
+### Not included:
+- Archive test bed integration
+- Subsequent FIS or IFRS updates for the upload path
+- Any front-end work that may be needed
+- GHGA-Connector update
+
+### Upload Controller Service (UCS)
+
+The UCS is responsible for two things:
+1. Cataloging individual files for a given study without knowing
+anything about studies or other business concepts.
+2. Facilitating the actual upload of said files by:
+   - Opening a new multipart S3 upload
+   - Dispensing pre-signed part upload URLs
+   - Completing/cancelling the S3 upload
+
+In the UCS, a study is represented by a minimalistic abstraction called a
+`FileUploadBox`, and a file is represented as a `FileUpload`. The former offers a way to
+group file uploads, lock them for changes and check the uniqueness of the included file aliases. The latter offers specific
+file information and a way for other services to be aware of that file's existence.
```

Comment:
I would not try to map this concepts to the GHGA domain too much. Especially the assumptions about the relationship to "study" won't hold, so this might be confusing to future readers.